db/postgres/tags.go

```diff
@@ -0,0 +1,92 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/satisfactorymodding/smr-api/util"
+
+	"github.com/finnbear/moderation"
+
+	"github.com/patrickmn/go-cache"
+)
+
+func ValidateTagName(tag string) error {
+	if len(tag) > 20 {
+		return errors.New("Tag name is over 20 characters long")
+	}
+	if moderation.IsInappropriate(tag) {
+		return errors.New("Tag name is inapproriate")
+	}
+	return nil
+}
+
+func CreateTag(tag *Tag, ctx *context.Context, ratelimit bool) (*Tag, error) {
+	err := ValidateTagName(tag.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	exists := GetTagByName(tag.Name, ctx)
```

Comment:
Should attempt to insert and check for response error if fails, no need to check before creating.